action_server/frontend/src/views/actions/components/ActionRun.tsx

```diff
@@ -177,7 +180,7 @@ export const ActionRun: FC<Props> = ({ action, actionPackage }) => {
           <Header size="small">
             <Header.Title title="Result" />
           </Header>
-          <Code lineNumbers={false} value={result.errorMessage || (result.data as string)} />
+          <Code lineNumbers={false} value={result.errorMessage || stringifyResult(result.data)} />
```

Comment:
Could be done by builtins maybe
```suggestion
          <Code lineNumbers={false} value={result.errorMessage || String(result.data ?? '')} />
```